### DISCUSSION TITLE
Bitmap.Clone Enhancement

### DISCUSSION THREAD

**Discussion by neoxeo** [alternetsoft/AlternetUI#136] [Ideas]
@generalloki 

Could you add Bitmap.Clone(Rectangle, PixelFormat) 
(https://learn.microsoft.com/en-us/dotnet/api/system.drawing.bitmap.clone?view=windowsdesktop-8.0#system-drawing-bitmap-clone(system-drawing-rectangle-system-drawing-imaging-pixelformat)

Thanks

**Comment by generalloki** [comment]
This method with PixelFormat parameter looks complex to implement as we need to map it to wxWidgets call. For what reason are you going to use it? What is the task?

**Comment by neoxeo** [comment]
@generalloki 

Here is the application I try to convert to AlternetUI : 
[https://github.com/tgraupmann/WinForm_Ollama_Copilot](https://github.com/tgraupmann/WinForm_Ollama_Copilot)

Here is the code used for OCR and it use "CopyFromScreen" and "Bitmap.Clone" : 
[https://github.com/tgraupmann/WinForm_Ollama_Copilot/blob/master/OcrManager.cs](https://github.com/tgraupmann/WinForm_Ollama_Copilot/blob/master/OcrManager.cs)

And here you can see how it works : 
[https://www.youtube.com/watch?v=WHM-Bc9k04E](https://www.youtube.com/watch?v=WHM-Bc9k04E)

**Comment by generalloki** [comment]
I would ask in other words: What PixelFormat values you would use? I don't see using of Bitmap.Clone there

**Comment by generalloki** [comment]
I suggest to use SkiaSharp and not wxWidgets based canvas as SkiaSharp is truly cross-platform and is available on all platforms including mobile devices.
